### PR TITLE
Add a window resize handler to reset image

### PR DESCRIPTION
**Pull request by alekseyg**
This takes care of resizing the window and rotating a mobile device. It
does reset the zoom as well, but in order to fix that, the zoom needs
to be fixed.

I found an issue with zoom, for which I will open a separate issue.

### CODE REVIEW

**Comment by alekseyg** [comment]
OK, I've added some more things. I've optimized the landscape view on small screens. Before doing that, I went ahead and flattened out the CSS, as nested styles are bad most of the time. They are less performant and make overriding a nightmare, among other things. I added a breakpoint for heights as well as widths to have it adapt to all kinds of screen sizes nicely. Then I made the controls hide by default on small screens.

That last thing is the only one I'm a bit concerned with. In order to toggle the controls when the image is tapped, I had to put the handler on the Image component, but the class name had to be toggled on the main lightbox. To do this, I simply went up the DOM tree and set the class, as I don't know if it is possible to access the parent component from the Image. There might be a better way to do this, but I don't know what it is.

File: src/classNames.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+const classNames = {
```
Comment (jfcaiceo) [inline src/classNames.js]: I prefer this file inside 'src/utils' folder.

**Comment by jfcaiceo** [comment]
> Before doing that, I went ahead and flattened out the CSS, as nested styles are bad most of the time.

Ok, it seems right.

> There might be a better way to do this, but I don't know what it is.

Right now, I think only in 2 other ways:
1. Make a function in the `Container` component, that with a `ref` to the lightbox-backdrop div, it toggle its class (I always prefer change a state or a props, but in this case, we can't because the `Image` component will render again with its default state). So then, we give that function to the `Image` as a prop.
2. Get that div with a vanilla javascript. There can only be one `Container` component active in the document, so the first occurrence is the one that you are looking for. Its easier but a little ugly.

What do you think?

**Comment by alekseyg** [comment]
I've moved `classNames` into `src/utils`.

Another way to do this probably will be to go ahead and put the tap handler into Container, even though that will create a bit of code duplication. One of the issues with the current solution is that it doesn't fire when the zoom/rotate/download controls are hidden. I don't like the fragile interdependence that is caused between the two components, so in this case a bit of code duplication would be the lesser of two evils. Of course, the two components are kind of interdependent, but that stems from the choice to make them into two components over making one huge monolithic component. Either way, avoiding fragility is probably the better choice here, now that I had a night to think about it.

**Comment by alekseyg** [comment]
Never mind that, I see your point. Changing the state on the container causes the Image to re-render, which in turn throws an error during unmounting as well. Hmm. The reason is because ~~zoom and pan are not stored in state~~ props override state when the component is re-rendered, if I understand correctly.

**Comment by alekseyg** [comment]
One way to make this easier is to move all the state and handlers to the container. This would make it huge, but that is solvable by composing the component out of several objects, with each taking care of a certain set of handlers or whatnot. The Container and Image components are very interdependent, and perhaps to simplify things, it might be better to move most of the behavior to the container and make the Image component more of a "dumb" component. This seems to be more in line with the way the React docs recommend to do things. For now, I'll just use method 1 you outlined. Before, I was using something akin to method 2.

**Comment by alekseyg** [comment]
OK, I added a function on `Container` and passed it as a prop to `Image` to take care of toggling controls.

**Comment by alekseyg** [comment]
I noticed an issue with tap detection and fixed it. This required using a different way to detect whether the mouse/finder is pressed for panning, and preserving the `startPoints` when the mouse/finger is released. No big deal, but more reliable tap detection now. Also, I made the download link stay white on hover like the rest of the controls.

**Comment by jfcaiceo** [comment]
It looks good!, thanks. Could you add a test to the tap detection too? It would be great. But it is ok now, and we can leave it to other PR.

**Comment by alekseyg** [comment]
Hmm, how to write a test for something (`Image`) that relies on a timeout and sets the class of its containing component (`Container`)...

I would imagine 3 tests to throw at it:
1. Long touch/click: Simulate `touchStart`; after more than 200ms, simulate `touchEnd`; expect the class `hide-controls` to not have been toggled on the `Container`.
2. Swipe: Simulate `touchStart` with one set of coordinates; immediately simulate `touchEnd` with different coordinates; after more than 200ms, expect the class `hide-controls` to not have been toggled on the `Container`.
3. Tap: Simulate `touchStart` with one set of coordinates; immediately simulate `touchEnd` with same coordinates; after more than 200ms, expect the class `hide-controls` to have been toggled on the `Container`.

How to actually write that into `imageSpec.js`, I don't know. Is it possible to put timeouts into tests?

**Comment by jfcaiceo** [comment]
It's possible with jasmine, but I can do it later.

**Pull request merged**